### PR TITLE
[csharp][aspnetcore] Allow the use of endpoints (single class per operation)

### PR DESCRIPTION
**Pull request by Blackclaws**
This adds support for endpoints, i.e. a single class per operation to the generator. Unfortunately the way the generator currently works they are still all part of the same file. We can discuss whether we want to put them under an even further nested namespace. 
As a side effect enabled actual usage of annotations. Might want to expand the annotations of the normal controllers later on as well.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@mandrean (2017/08) @frankyjuang (2019/09) @shibayan (2020/02)

### CODE REVIEW

**Comment by wing328** [comment]
cc @lucamazzanti as well (new C# technical committee member)

**Review by lucamazzanti — COMMENTED** [review]
The namespace doesn't respect the folders, are you sure to keep the .Endpoints part?
Classes doesn't respect the Controller suffix convention, do you prefer to remove it because it can became a long name?

**Comment by Blackclaws** [comment]
> The namespace doesn't respect the folders, are you sure to keep the .Endpoints part?
> Classes doesn't respect the Controller suffix convention, do you prefer to remove it because it can became a long name?

Good point, maybe we should actually change the folder instead? I think I want to keep the Endpoints part because I want to clearly separate it from the existing functionality regarding controllers. 

In general I would also like to have one file per operation however the way the generator works right now this simply isn't possible and would require a new option for the template system that allows creating one file per operation instead of per tag.

The Controller suffix convention is just that a convention and we are doing route generation via annotations anyway. I was thinking about actually adding an "Endpoint" Suffix but that might be redundant for people that already call it Endpoint in their apispec. Now that I\m thinking about it that might be something we can parse out however .... 

When you go by how other people are implementing a similar feature you can end up with even more different classes/names.

https://github.com/ardalis/CleanArchitecture/tree/main/src/Clean.Architecture.Web/Endpoints/ProjectEndpoints

Basically the structure would be: Endpoints.TAGEndpoints followed by classes that are named by the operationid

**Comment by wing328** [comment]
i've filed https://github.com/OpenAPITools/openapi-generator/pull/17161 to add a new rule in openapi-normalizer to set `tags` to the operationId so as to support single "class" per operation (or per operationId) to be more precise.

Please take a look when you've time.

**Comment by wing328** [comment]
#17161 merged. please pull the latest master to give it a try. closing this one.
